### PR TITLE
ci: lint only changed files when safe

### PR DESCRIPTION
**Pull request by silvenon**
<br/><br/><br/><url>LiveURL: https://orbit-ci-improve-speed-eslint.surge.sh</url>

### CODE REVIEW

**Comment by silvenon** [comment]
`fancy-log` is used for `gulp` output, that's why I'm using it.

**Comment by silvenon** [comment]
There appears to be a problem with `@adeira/monorepo-utils`'s git command on CI, perhaps it's using a different version.

File: tasks/eslintCI.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  } else {
+    filesToLint = changedFiles.filter(fileName => fileName.match(/\.(jsx?|tsx?|flow)$/));
+  }
+
+  const results = await eslint.lintFiles(filesToLint);
+  const errorResults = ESLint.getErrorResults(results); // --quiet
+
+  if (results.errorCount > 0) {
+    const formatter = await eslint.loadFormatter();
+    console.log(formatter(errorResults));
```
Comment (mainframev) [inline tasks/eslintCI.js:42]: maybe here we can use `log.error(...)` from `fancy-log` as we already imported that module :) Or it does not make sense to use with `eslint.loadFormatter()`? 🤔 

Everything works good :)

**Review by mainframev — DISMISSED** [review]

File: tasks/eslintCI.js (hunk 1 above)
Comment (silvenon) [reply tasks/eslintCI.js:42]: Thanks for the suggestion, but I think it's best to use `fancy-log` for single line outputs, this will span multiple lines.

**Comment by silvenon** [comment]
This works now, please review again because the PR had changed significantly.

**Review by mainframev — APPROVED** [review]

**Pull request merged**